api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -739,4 +760,44 @@ private void createConfigFile(org.pmiops.workbench.firecloud.model.Workspace fcW
     response.setItems(workspaces.stream().map(TO_CLIENT_WORKSPACE).collect(Collectors.toList()));
     return ResponseEntity.ok(response);
   }
+
+  private String convertClusterName(String workspaceId) {
+    String clusterName = workspaceId + this.userProvider.get().getUserId();
+    clusterName = clusterName.toLowerCase();
+    return clusterName;
+  }
+
+  /**
+   * Gets the notebook files from google bucket.
+   * In case origin is createCluster, add config files to the file list and return
+   * @param workspaceNamespace
+   * @param workspaceId
+   * @param directory
+   * @return List of FileDetail
+   * @throws org.pmiops.workbench.firecloud.ApiException
+   */
+  private List<FileDetail> getFilesFromBucket(String workspaceNamespace,
```

Comment:
So if I'm reading this method right, it always returns notebooks but if directory is set to CONFIG_WORKSPACE_DIRECTORY it also returns config... is that right?

That's a bit confusing. My initial assumption was "directory" would be the only place we'd look for files... and it's not clear that special logic will be applied for filtering notebooks but not config...

Could we instead have something like:

getFilesFromBucket(workspaceNamespace, workspaceId, directory, filenamePredicate)

and call it twice, once for notebooks and once for the config? Or have separate methods getNotebookFilesFromBucket() and getConfigFilesFromBucket()? Then you could concatenate the resulting lists together.

Reply:
Yes it was getting the Notebooks as well as config files based on the directory information. Since notebooks will have the check for ipynb and then the the logic for conversion of bucket filename to filedetails is common for both, i combined both the calls in one, just to not repeat the fileDetails conversion method. I can do separate methods and combining them to make it cleaner

Reply:
Fixed. There are two methods each dedicated to one folder files each.